app/javascript/controllers/disable_controller.js

```diff
@@ -0,0 +1,39 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  connect() {
+    this.element.dataset.action = `${this.element.dataset.action || ""} submit->disable#disableForm`
```

Comment:
i see that this controller is added to `_form`. is there a reason for adding the action here instead of where the controller is defined? is this because the `dataset` may not be present yet or something? (in the case of a new upload, for example?)

Reply:
So this is a stimulus trick I've seen in the examples lately...

Instead of this:

```html
<form data-controller="disable" data-action="submit->disable#disable">
```

We can just write:

```html
<form data-controller="disable">
```

And then on `connect()` ... the disable controller adds that action to the controller itself.  Ends up being the same as the 1st example, saving a few characters.  And you don't have to mess with `addEventListener` / `removeEventListener` manually.  Stimulus still handles it.

Reply:
(And yes, I'm trying to handle the case where the DOM already has some other action in it ... I don't want to nuke that, adding myself to the data-actions).